client/views/admin/invites/InviteRow.tsx

```diff
@@ -1,25 +1,43 @@
 import { Button, Icon, Table, Box } from '@rocket.chat/fuselage';
 import { useMediaQuery } from '@rocket.chat/fuselage-hooks';
 import moment from 'moment';
-import React from 'react';
+import React, { ReactElement, MouseEvent } from 'react';
 
-import { useModal } from '../../../contexts/ModalContext';
+import { IInvite } from '../../../../definition/IInvite';
+import GenericModal from '../../../components/GenericModal';
 import { useEndpoint } from '../../../contexts/ServerContext';
 import { useToastMessageDispatch } from '../../../contexts/ToastMessagesContext';
 import { useTranslation } from '../../../contexts/TranslationContext';
 import { useFormatDateAndTime } from '../../../hooks/useFormatDateAndTime';
+import { imperativeModal } from '../../../lib/imperativeModal';
 
-function InviteRow({ _id, createdAt, expires, days, uses, maxUses, onRemove }) {
+type InviteRowProps = IInvite & {
+	onRemove: (_id: IInvite['_id']) => void;
+};
+
+const InviteRow = ({
+	_id,
+	createdAt,
+	expires,
+	days,
+	uses,
+	maxUses,
+	onRemove,
+}: InviteRowProps): ReactElement => {
 	const t = useTranslation();
 	const formatDateAndTime = useFormatDateAndTime();
-	const modal = useModal();
 	const dispatchToastMessage = useToastMessageDispatch();
-
 	const removeInvite = useEndpoint('DELETE', `removeInvite/${_id}`);
 
-	const daysToExpire = ({ expires, days }) => {
+	const daysToExpire = ({
+		expires,
+		days,
+	}: {
+		expires: IInvite['expires'];
+		days: IInvite['days'];
+	}): string => {
 		if (days > 0) {
-			if (expires < Date.now()) {
+			if (expires && expires.getTime() < new Date().getTime()) {
```

Comment:
maybe decoupling a method isExpired here. Is possible to get days null or undefined?

Reply:
It's possible to get null